spring-data-cassandra/src/main/java/org/springframework/data/cassandra/config/CqlSessionFactoryBean.java

```diff
@@ -450,11 +451,20 @@ public void afterPropertiesSet() {
 
 		this.session = buildSession(sessionBuilder);
 
-		executeCql(getStartupScripts().stream(), this.session);
-		performSchemaAction();
+		try {
+			SchemaRefreshUtils.withDisabledSchema(this.session, () -> {
+				executeCql(getStartupScripts().stream(), this.session);
+				performSchemaAction();
+			});
+		} catch (RuntimeException e) {
+			throw e;
+		} catch (Exception e) {
+			throw new IllegalStateException("Unexpected checked exception thrown", e);
+		}
 
-		this.systemSession.refreshSchema();
-		this.session.refreshSchema();
+		if (this.systemSession.isSchemaMetadataEnabled()) {
+			this.systemSession.refreshSchema();
```

Comment:
The schema refresh in `SessionFactoryFactoryBean` should be guarded as well.

Reply:
I believe that's already tackled in this PR, let me know if I missed something!